odoo/api.py

```diff
@@ -918,41 +902,35 @@ def check(self, env):
         # flush fields to be recomputed before evaluating the cache
         env['res.partner'].recompute()
 
-        # make a full copy of the cache, and invalidate it
-        dump = defaultdict(dict)
-        key_cache = self._data
-        for field, field_cache in key_cache.items():
-            for record_id, value in field_cache.items():
-                if record_id:
-                    dump[field][record_id] = value
-
+        # make a copy of the cache, and invalidate it
+        dump = dict(self._data)
```

Comment:
any reason why it was done manually before?

Reply:
The purpose was to skip the check for new records.  I just re-introduced this condition in the actual check below.